File: clients/java/examples/collections-find-one-and-update-after.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.Collection;
import com.datastax.astra.client.collections.commands.ReturnDocument;
import com.datastax.astra.client.collections.commands.Update;
import com.datastax.astra.client.collections.commands.options.CollectionFindOneAndUpdateOptions;
import com.datastax.astra.client.collections.definition.documents.Document;
import com.datastax.astra.client.core.query.Filter;
import com.datastax.astra.client.core.query.Filters;
import java.util.Optional;

public class Example {

  public static void main(String[] args) {
    // Get an existing collection
    Collection<Document> collection =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getCollection("**COLLECTION_NAME**");

    // Update a document
    Filter filter = Filters.eq("_id", "101");
    Update update = Update.create().set("color", "blue");
    CollectionFindOneAndUpdateOptions options =
        new CollectionFindOneAndUpdateOptions().returnDocument(ReturnDocument.AFTER);
    Optional<Document> result = collection.findOneAndUpdate(filter, update, options);
    System.out.println(result);
  }
}

File: clients/python/examples/collections-find-one-and-update-after.py
from astrapy import DataAPIClient
from astrapy.constants import ReturnDocument

# Get an existing collection
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
collection = database.get_collection("**COLLECTION_NAME**")

# Update a document
result = collection.find_one_and_update(
    {"_id": "101"},
    {"$set": {"color": "blue"}},
    return_document=ReturnDocument.AFTER,
)

print(result)
